api-sdk/src/main/java/com/smartling/api/sdk/FileApiClientAdapterImpl.java

```diff
@@ -230,44 +236,90 @@ public StringResponse getFile(final GetFileParameterBuilder getFileParameterBuil
 
         return apiResponse;
     }
 
-    private HttpPost createFileUploadHttpPostRequest(final String apiParameters, final ContentBody contentBody)
-    {
-        final MultipartEntityBuilder multipartEntityBuilder = MultipartEntityBuilder.create()
-                .addPart(FileApiParams.FILE, contentBody);
-
-        final HttpPost httpPost = new HttpPost(String.format(UPLOAD_FILE_API_URL, baseApiUrl) + apiParameters);
-        httpPost.setEntity(multipartEntityBuilder.build());
-
-        return httpPost;
-    }
-
     private ApiResponse<UploadFileData> uploadFile(final FileUploadParameterBuilder fileUploadParameterBuilder, final ContentBody contentBody)
             throws ApiException
     {
         logger.debug(String.format("Upload file: fileUri = %s, projectId = %s, apiKey = %s, localesToApprove = %s",
-                fileUploadParameterBuilder.getFileUri(), this.projectId, maskApiKey(this.apiKey), StringUtils.join(fileUploadParameterBuilder.getLocalesToApprove(), ", ")));
+                        fileUploadParameterBuilder.getFileUri(), this.projectId, maskApiKey(this.apiKey), StringUtils.join(fileUploadParameterBuilder.getLocalesToApprove(), ", ")));
 
         final List<NameValuePair> paramsList = fileUploadParameterBuilder.getNameValueList();
         final String params = buildParamsQuery(paramsList.toArray(new NameValuePair[paramsList.size()]));
         final HttpPost httpPostFile = createFileUploadHttpPostRequest(params, contentBody);
 
-        final StringResponse response = getHttpUtils().executeHttpCall(httpPostFile, proxyConfiguration);
-        final ApiResponse apiResponse = getApiResponse(response.getContents(), new TypeToken<ApiResponseWrapper<UploadFileData>>() {});
+        final ApiResponse<UploadFileData> apiResponse = getResponse(httpPostFile, new TypeToken<ApiResponseWrapper<UploadFileData>>() {});
         logger.debug(String.format("Upload file: %s. %s", apiResponse.getCode(), getApiResponseMessages(apiResponse)));
 
         return apiResponse;
     }
 
+    private <T extends Data> ApiResponse<T> getResponse(final HttpRequestBase httpRequest, final TypeToken<ApiResponseWrapper<T>> typeToken) throws ApiException
+    {
+        final StringResponse response = getStringResponse(httpRequest);
+
+        return parseApiResponse(response.getContents(), typeToken);
+    }
+
+    private StringResponse getStringResponse(final HttpRequestBase httpRequest) throws ApiException
+    {
+        StringResponse stringResponse = getHttpUtils().executeHttpCall(httpRequest, proxyConfiguration);
+
+        if (stringResponse.isSuccess()) {
+            return stringResponse;
+        }
+
+        String contents = stringResponse.getContents();
+        logger.error(String.format("Non-successful response: \n contents: %s", contents));
+        throw createApiException(contents);
+    }
+
+    private ApiException createApiException(final String contents)
```

Comment:
I would rename to `newApiException`... Just because "throw new API exception" sounds natural